### ISSUE TITLE
TypeScript signatures

### ISSUE THREAD

**Issue by haf** [danielr18/connected-next-router#22]
It would help consuming this library if there were typescript signatures available as part of the package.

**Comment by danielr18** [comment]
I don't have much experience with typescript, will look into it. If anyone can help with a PR however, that would be nice.

**Comment by haf** [comment]
@danielr18 Thank you. I don't have much experience with publishing node modules with typescript either, but I think the easiest way to add them, is to start adding them inline in the code, like described here https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html

**Comment by haf** [comment]
https://github.com/haf/connected-next-router/tree/typescript

Compiles and should be ready to publish; but since there are no tests in your repository nor any examples, I'm not that confident everything works.

**Comment by danielr18** [comment]
Thank you! I'll be checking this out today

**Comment by danielr18** [comment]
By the way, given that as you mentioned, it's not based on the existing history, it would be hard to make a PR, I can make the PR based on your fork, but you wouldn't show up as a [contributor](https://github.com/danielr18/connected-next-router/graphs/contributors). Is that important for you?

**Comment by haf** [comment]
Not important at all. Add my name to the list of authors if you want.

I think this one should work when published, but I can't make npm run 'prepare' when referencing from git.

**Comment by haf** [comment]
Example for current branch:

```
import { LocationChangedAction } from 'connected-next-router/dist/jsm/actions'
// ..
import { LOCATION_CHANGE } from 'connected-next-router'
```

**Comment by haf** [comment]
Did it go well?

**Comment by danielr18** [comment]
I just published version: 2.0.0-alpha.1, check the [typescript branch](https://github.com/danielr18/connected-next-router/tree/typescript) for a copy of the basic example using typescript.

It has a couple of breaking changes that still need to be documented, like supporting only Next.js 9 and react 16.8.0. Let me know if the alpha version works well in your project, any feedback is welcome!

**Comment by haf** [comment]
Did you get around to typing the actions? I was planning on testing your branch tomorrow.

**Comment by nitindhar7** [comment]
Any updates on this?

**Comment by danielr18** [comment]
@nitindhar7 I haven't released v2.0.0 yet which has TypeScript signatures, but the latest version in NPM is the release candidate (2.0.0-alpha.3). I suggest you try that and comment here if it's working well for your use case.

**Comment by danielr18** [comment]
Released on v3
